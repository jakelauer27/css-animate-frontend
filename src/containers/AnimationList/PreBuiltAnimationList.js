import React, {Component} from 'react'
import { connect } from 'react-redux'
import { uid } from 'react-uid'
import { PropTypes } from 'prop-types'

export class PreBuiltAnimationList extends Component {

  selectAnimation(e) {
    const items = document.querySelectorAll('.animation-li')
    items.forEach(item => item.classList.remove('selected-animation'))
    e.target.classList.add('selected-animation')
  }

  render() {
    const { prebuiltAnimations, currentAnimation } = this.props
    return (
      <section className='prebuilt-animation-section'>
        <ul className= 'prebuilt-animation-list'>
        {
          prebuiltAnimations.map(animation => {
            let selected = ''
            if (animation.id === currentAnimation.id) {selected = 'selected-animation'}
            return (
              <li 
                className={`animation-li ${animation.id} ${selected}`} 
                key={uid(animation)}
                onClick={e => this.selectAnimation(e) }>
               {animation.ani_name}
              </li>
            )
          })
        }
        </ul>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  prebuiltAnimations: state.prebuiltAnimations,
  currentAnimation: state.currentAnimation
})

PreBuiltAnimationList.propTypes = {
  prebuiltAnimations: PropTypes.array,
  currentAnimation: PropTypes.object
}

export default connect(mapStateToProps)(PreBuiltAnimationList)