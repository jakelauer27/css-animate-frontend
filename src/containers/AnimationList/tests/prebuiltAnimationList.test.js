import React from 'react'
import { shallow } from 'enzyme'
import { PreBuiltAnimationList, mapStateToProps } from '../PreBuiltAnimationList'

describe('PreBuiltAnimationList', () => {
  let mockEvent
  let wrapper
  const mockAnimations = [{id: 1, 'ani-name': 'slideIn' },{id: 2, 'ani-name': 'rotateIn' }]
  const mockCurrentAnimation = {id: 2, 'ani-name': 'rotateIn' }

  beforeEach(() => {
    mockEvent = {target: {classList: ['2', 'test2']}}
    mockEvent.target.classList.add = jest.fn()
    mockEvent.target.classList.remove = jest.fn()
    wrapper = shallow(<PreBuiltAnimationList prebuiltAnimations={mockAnimations} currentAnimation={mockCurrentAnimation} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should run selectAnimation when an animation item is clicked', () => {
    const singleItemMockAnimations = [{id: 1, 'ani-name': 'slideIn' }]
    wrapper = shallow(<PreBuiltAnimationList prebuiltAnimations={singleItemMockAnimations} currentAnimation={mockCurrentAnimation} />)

    const spy = jest.spyOn(wrapper.instance(), 'selectAnimation')
    wrapper.find('.animation-li').simulate('click', mockEvent)
    expect(spy).toHaveBeenCalled()
  })

  describe('selectAnimation', () => {

    it('should remove selected-animation class from all nodes', () => {
      document.querySelectorAll = jest.fn().mockImplementation(() => {
        return [{classList: {remove: jest.fn()}}]
      })
      wrapper.instance().selectAnimation(mockEvent)

      expect(document.querySelectorAll).toHaveBeenCalledWith('.animation-li')
    })
  })

  describe('mapStateToProps', () => {

    it('should return a props object with a currentAnimation and preBuiltAnimations property', () => {
      const mockState = {prebuiltAnimations: {keyframes: {name: 'slideInX'}}, user:{id: 1}, currentAnimation: {name: 'test'}}
      const expected = {prebuiltAnimations: {keyframes: {name: 'slideInX'}}, currentAnimation: {name: 'test'}}
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })
})