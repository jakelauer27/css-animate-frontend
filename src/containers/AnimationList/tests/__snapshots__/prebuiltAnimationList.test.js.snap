// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreBuiltAnimationList should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreBuiltAnimationList
    currentAnimation={
      Object {
        "ani-name": "rotateIn",
        "id": 2,
      }
    }
    prebuiltAnimations={
      Array [
        Object {
          "ani-name": "slideIn",
          "id": 1,
        },
        Object {
          "ani-name": "rotateIn",
          "id": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="prebuilt-animation-list"
      >
        <li
          className="animation-li 1 "
          onClick={[Function]}
        />
        <li
          className="animation-li 2 selected-animation"
          onClick={[Function]}
        />
      </ul>,
      "className": "prebuilt-animation-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="animation-li 1 "
            onClick={[Function]}
          />,
          <li
            className="animation-li 2 selected-animation"
            onClick={[Function]}
          />,
        ],
        "className": "prebuilt-animation-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "animation-li 1 ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "uid2",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "animation-li 2 selected-animation",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="prebuilt-animation-list"
        >
          <li
            className="animation-li 1 "
            onClick={[Function]}
          />
          <li
            className="animation-li 2 selected-animation"
            onClick={[Function]}
          />
        </ul>,
        "className": "prebuilt-animation-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="animation-li 1 "
              onClick={[Function]}
            />,
            <li
              className="animation-li 2 selected-animation"
              onClick={[Function]}
            />,
          ],
          "className": "prebuilt-animation-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "uid1",
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "animation-li 1 ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "uid2",
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "animation-li 2 selected-animation",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
